collect_app/src/main/java/org/odk/collect/android/external/FormUriActivity.kt

```diff
@@ -32,44 +57,155 @@ class FormUriActivity : ComponentActivity() {
         super.onCreate(savedInstanceState)
         DaggerUtils.getComponent(this).inject(this)
 
-        if (savedInstanceState != null) {
-            formFillingAlreadyStarted = savedInstanceState.getBoolean(FORM_FILLING_ALREADY_STARTED)
+        when {
+            !assertProjectListNotEmpty() -> Unit
```

Comment:
Theoretically yes but for example `assertFormExists` displays a dialog with different messages depending on the cause of the issue... I would need to divide it into a few smaller methods which again theoretically would be nice but in this case, doesn't make sense since we are checking a few following steps in sequence.